cylc/rose/utilities.py

```diff
@@ -189,10 +189,15 @@ def rose_config_exists(
 
     # If _any_ of the following are true we want to return True.
     if (
-        Path(srcdir, 'rose-suite.conf').is_file() or
-        opts and opts.opt_conf_keys or
-        opts and opts.defines or
-        opts and opts.rose_template_vars
+        (
+            getattr(opts, 'opt_conf_keys', False)
+        ) or (
+            opts and hasattr(opts, 'defines')
+            and opts.defines
+        ) or (
+            opts and hasattr(opts, 'rose_template_vars')
+            and opts.rose_template_vars
+        )
     ):
```

Comment:
This block can still be tidied up - but I'm not sure you've resolved Oliver's point abut it: https://github.com/cylc/cylc-rose/pull/92#discussion_r746726875